ComponentKit/Core/CKDimension.mm

```diff
@@ -131,12 +131,11 @@ _Range intersect(const _Range &other) const
 }
 size_t CKSizeRange::hash() const
 {
-  std::hash<CGFloat> hasher;
   NSUInteger subhashes[] = {
-    hasher(min.width),
-    hasher(min.height),
-    hasher(max.width),
-    hasher(max.height)
+      NSUInteger(min.width),
+      NSUInteger(min.height),
+      NSUInteger(max.width),
+      NSUInteger(max.height)
```

Comment:
Hi Adamjernst,
It does have little difference in precision. Let’s narrow down our comparison in iPhone simulator with the following two cases.
NSLog(@"%ld", CKSizeRange(CGSizeMake(50, 90), CGSizeMake(60, 100)).hash());
NSLog(@"%ld", CKSizeRange(CGSizeMake(50, 91), CGSizeMake(60, 100)).hash());

The only difference above is the min size range with 90 or 91 in width. Then let’s try our first configuration A:
- Have hasher() to construct subhashes
- Do bit shift with sizeof(unsigned long long)

And we get the following results and can’t differentiate our test samples.
2015-04-24 15:18:37.060 WildeGuess[2330:230394] -3577828428969148416
2015-04-24 15:18:37.060 WildeGuess[2330:230394] -3577828428969148416

So let’s try the second configuration B:
- Have NSUInteger to construct subhashes
- Do bit shift with sizeof(NSUInteger)

And we get the following results and can differentiate our test samples.
2015-04-24 15:20:54.276 WildeGuess[2372:237574] 844774500
2015-04-24 15:20:54.277 WildeGuess[2372:237574] 844840036

Also I found another thing. For example, if we search keyword "CKIntegerArrayHash" in Xcode, we can have two groups of results. Results in ComponentKit adopt NSUInteger to construct subhashes (ex. CKComponentDataSourceInputItem); however, results in ComponentTextKit adopt std::hash to construct subhashes (ex. CKTextKitRendererCache).

My assumption is that NSUInteger is more native to compiler and Apple pays more efforts on it for cast or internal conversion. In simple way to describe this, you can say I’m thinking CGFloat->NSInteger->std::hash is potentially better than CGFloat->std::hash directly. In order to avoid the situation of mixing up C++/C and type cast multiple times, I created another version of CKIntegerArrayHash in pure Swift and do another comparison . I paste the code here:

```
func CKIntegerArrayHash <H: Hashable, S: SequenceType where S.Generator.Element == H>(sequence: S, count: Int) -> Int {
    var result = 0
    for element in sequence {
        result = result.hashValue << sizeof(Int) | element.hashValue
    }
    return result
}
```
The result I got is as same as to the results of configuration B. 
Therefore, I chose to cast NSUInteger and sizeof(NSUInteger) to remind me I’m dealing with NSUInteger.

Cheers,

Allen